docs/networking/troubleshooting.md

```diff
@@ -0,0 +1,111 @@
+# Troubleshooting
+
+`coder ping <workspace>` will ping the workspace agent and print diagnostics on
+the state of the connection. These diagnostics are created by inspecting both
+the client and agent network configurations, and provide insights into why a
+direct connection may be impeded, or why the quality of one might be degraded.
+
+The `-v/--verbose` flag can be appended to the command to print client debug
+logs.
+
+```console
+$ coder ping dev
+pong from workspace proxied via  DERP(Council Bluffs, Iowa)  in 42ms
+pong from workspace proxied via  DERP(Council Bluffs, Iowa)  in 41ms
+pong from workspace proxied via  DERP(Council Bluffs, Iowa)  in 39ms
+✔ preferred DERP region: 999 (Council Bluffs, Iowa)
+✔ sent local data to Coder networking coordinator
+✔ received remote agent data from Coder networking coordinator
+    preferred DERP region: 999 (Council Bluffs, Iowa)
+    endpoints: x.x.x.x:46433, x.x.x.x:46433, x.x.x.x:46433
+✔ Wireguard handshake 11s ago
+
+❗ You are connected via a DERP relay, not directly (p2p)
+Possible client-side issues with direct connection:
+ - Network interface utun0 has MTU 1280, (less than 1378), which may degrade the quality of direct connections
+
+Possible agent-side issues with direct connection:
+ - Agent is potentially behind a hard NAT, as multiple endpoints were retrieved from different STUN servers
+ - Agent IP address is within an AWS range (AWS uses hard NAT)
+```
+
+## Common Problems
```

Comment:
It's not clear from reading this that these are problems specifically with establishing direct connections, not when failing to establish connectivity at all.

Reply:
That's fair, I'll leave it at `Common Problems with Direct Connections`.